### PR TITLE
Add filler data in statistics

### PR DESCRIPTION
**Pull request by sctanf**
Make statistics continue to show server-side data for a while if client fails to send statistics
Also send "redraws" to reduce judder in statistics graph

### CODE REVIEW

File: alvr/server/cpp/alvr_server/ClientConnection.cpp [hunk 1]
```diff
@@ -263,6 +267,81 @@ void ClientConnection::ProcessRecv(unsigned char *buf, size_t len) {
...
 		}
 	}
+	FillStatistics();
+}
+
+void ClientConnection::FillStatistics() {
+	uint64_t now = GetTimestampUs();
+	if (now - m_LastStatisticsFillerUpdate > STATISTICS_FILLER_US && now - m_LastStatisticsUpdate > STATISTICS_FILLER_US && now - m_LastStatisticsUpdate < STATISTICS_TIMEOUT_US)
+	{
+		//timings might be a little incorrect since it is a mix from a previous sent frame and latest frame
```
Comment (ckiee) [inline alvr/server/cpp/alvr_server/ClientConnection.cpp:277]: Do we really want to show any timing info then? Providing bad data is arguably worse than no data until the client is reconnected.

**Review by ckiee — COMMENTED** [review]
I am scared by the JavaScript, but it seems fine. (although I haven't tested it locally)

File: alvr/server/cpp/alvr_server/ClientConnection.cpp (hunk 1 above)
Comment (sctanf) [reply alvr/server/cpp/alvr_server/ClientConnection.cpp:277]: I did not change the comment, that should be for rendered frames
Frames are still rendered while the client connection is broken or lagging (server fps reflects this)
I can remove it if the timings should only matter for frames sent to the client, though

Comment (ckiee) [reply alvr/server/cpp/alvr_server/ClientConnection.cpp:277]: Resolving as it's better to keep this PR isolated to one thing.

**Comment by zarik5** [comment]
@sctanf do you think this can be implemented purely in javascript?

**Comment by sctanf** [comment]
@zarik5 sure

**Comment by sctanf** [comment]
Filler data no longer contains updated server statistics, only repeating some last sent data for up to 1 second

**Review by zarik5 — APPROVED** [review]

**Pull request merged**